modules/library/referencing/src/main/java/org/geotools/referencing/operation/matrix/Matrix3.java

```diff
@@ -171,9 +134,9 @@ public void setMatrix(final AffineTransform transform) {
      * @since 2.3
      */
     public boolean equalsAffine(final AffineTransform transform) {
-        return m00==transform.getScaleX() && m01==transform.getShearX() && m02==transform.getTranslateX() &&
-               m10==transform.getShearY() && m11==transform.getScaleY() && m12==transform.getTranslateY() &&
-               m20==0                     && m21==0                     && m22==1;
+        return mat.data[0]==transform.getScaleX() && mat.data[1]==transform.getShearX() && mat.data[2]==transform.getTranslateX() &&
+               mat.data[3]==transform.getShearY() && mat.data[4]==transform.getScaleY() && mat.data[5]==transform.getTranslateY() &&
+               mat.data[6]==0                     && mat.data[7]==0                     && mat.data[8]==1;
```

Comment:
As a note, clone() doesn't work.
scala> val m = new Matrix3(1.0, 1.0, 1.0,1.0, 1.0, 1.0,1.0, 1.0, 1.0)
scala> m.clone
java.lang.ClassCastException: org.geotools.referencing.operation.matrix.GeneralMatrix cannot be cast to org.geotools.referencing.operation.matrix.Matrix3
	at org.geotools.referencing.operation.matrix.Matrix3.clone(Matrix3.java:164)